File: examples/DesignPatterns/FactoryMethod/src/main/java/reducers/impl/StringReducer.java
package reducers.impl;

import dispatchers.Dispatcher;
import dispatchers.impl.StringDispatcher;
import reducers.ReducerCreator;

import java.util.Random;

public class StringReducer extends ReducerCreator<String> {
    private final Random rand = new Random();
    @Override
    public Dispatcher<String> createDispatcher() {
        return new StringDispatcher(rand.nextInt(255));
    }
}

File: examples/DesignPatterns/FactoryMethod/typescript/src/reducers/impl/StringReducer.ts
import { random } from 'lodash';
import { Dispatcher, StringDispatcher } from '../../dispatchers';
import { Reducer } from '../Reducer';

export class StringReducer extends Reducer<string> {
  public createDispatcher(): Dispatcher<string> {
    return new StringDispatcher(random(1, 255));
  }
}
